src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,776 @@
+			{
+				var treeView = AncestorTreeView;
+				var key = e.Key;
+				var originalKey = e.OriginalKey;
+
+				if (originalKey == VirtualKey.GamepadA &&
+					treeView.ListControl.IsMultiselect)
+				{
+					HandleGamepadAInMultiselectMode(targetNode);
+					e.Handled = true;
+				}
+				else if (IsInReorderMode(key))
+				{
+					HandleReorder(key);
+					e.Handled = true;
+				}
+				// Collapse/Expand
+				else if (IsExpandCollapse(key))
+				{
+					var handled = HandleExpandCollapse(key);
+					if (handled)
+					{
+						e.Handled = true;
+					}
+				}
+				else if (key == VirtualKey.Space &&
+						 treeView.ListControl.IsMultiselect)
+				{
+					var selectionBox = m_selectionBox;
+					bool isSelected = CheckBoxSelectionState(selectionBox) == TreeNodeSelectionState.Selected;
+					selectionBox.IsChecked = !isSelected;
+					e.Handled = true;
+				}
+			}
+
+			// Call to base for all other key presses
+			base.OnKeyDown(e);
+		}
+
+		protected override void OnDrop(Windows.UI.Xaml.DragEventArgs args)
+		{
+			if (!args.Handled && args.AcceptedOperation == DataPackageOperation.Move)
+			{
+				TreeViewItem droppedOnItem = this;
+				var treeView = AncestorTreeView;
+				if (treeView != null)
+				{
+					var treeViewList = treeView.ListControl;
+					var droppedNode = treeViewList.DraggedTreeViewNode;
+					if (droppedNode != null)
+					{
+						if (treeViewList.IsMutiSelectWithSelectedItems)
+						{
+							var droppedOnNode = treeView.NodeFromContainer(droppedOnItem);
+							var selectedRoots = treeViewList.GetRootsOfSelectedSubtrees();
+							foreach (var node in selectedRoots)
+							{
+								var nodeIndex = treeViewList.FlatIndex(node);
+								if (treeViewList.IsFlatIndexValid(nodeIndex))
+								{
+									treeViewList.RemoveNodeFromParent(node);
+									((TreeViewNodeVector)droppedOnNode.Children).Append(node);
+								}
+							}
+
+							args.Handled = true;
+							treeViewList.OnDropInternal(args);
+						}
+						else
+						{
+							var droppedOnNode = treeView.NodeFromContainer(droppedOnItem);
+
+							// Remove the item that was dragged
+							int removeIndex = droppedNode.Parent.Children.IndexOf(droppedNode);
+
+							if (droppedNode != droppedOnNode)
+							{
+								((TreeViewNodeVector)droppedOnNode.Parent.Children).RemoveAt(removeIndex);
+
+								// Append the dragged dropped item as a child of the node it was dropped onto
+								((TreeViewNodeVector)droppedOnNode.Children).Append(droppedNode);
+
+								// If not set to true then the Reorder code of listview will override what is being done here.
+								args.Handled = true;
+								treeViewList.OnDropInternal(args);
+							}
+							else
+							{
+								args.AcceptedOperation = DataPackageOperation.None;
+							}
+						}
+					}
+				}
+			}
+			base.OnDrop(args);
+		}
+
+		protected override void OnDragOver(Windows.UI.Xaml.DragEventArgs args)
+		{
+			var treeView = AncestorTreeView;
+			if (treeView != null && !args.Handled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)